src/core/ddsi/src/ddsi_cdrstream_write.part.c

```diff
@@ -43,6 +51,16 @@ static void dds_stream_write_stringBO (DDS_OSTREAM_T * __restrict os, const char
     dds_os_put1BO (os, 0);
 }
 
+static bool dds_stream_write_bool_arrBO (DDS_OSTREAM_T * __restrict os, const uint8_t * __restrict addr, uint32_t num)
```

Comment:
It is a bit sad that it ends up like this when one'd really want to a `memcpy`. It might actually not be that much worse to do it like this so let's leave it for now.